llama_index/ingestion/pipeline.py

```diff
@@ -337,6 +341,15 @@ def _handle_upserts(
 
         return nodes_to_run
 
+    @staticmethod
+    def _node_batcher(
+        num_batches: int, nodes: Union[List[BaseNode], List[Document]]
+    ) -> Generator[Union[List[BaseNode], List[Document]], Any, Any]:
+        """Yield successive n-sized chunks from lst."""
+        batch_size = max(1, int(len(nodes) / num_batches))
+        for i in range(0, len(nodes), batch_size):
+            yield nodes[i : i + batch_size]
+
     def run(
```

Comment:
Is it possible to also add this functionality to `arun()` ?

Reply:
Yeah I looked into that a little bit. There is a `starmap_async` but that's a non-blocking version of `starmap` which I don't think is uses for parallelizing async tasks.

I can do so more digging tho on this and try to add a parallel processing of async tasks for us in this PR. I did coime across `ProcessPoolExecutor` of the `concurrent.futures` module as well as perhaps using asyncio. Let me see what we can do here.

Reply:
Yea if its a ton of work, we can introduce in another PR. But would be helpful to have an async entry point for people running on servers 🤔

Reply:
That's a good use case. I think I have it working -- and it's pretty fast :)